homeassistant/components/trafikverket_weatherstation/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Adds config flow for Trafikverket Weather integration."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_MONITORED_CONDITIONS, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+
+SENSOR_TYPES: dict[str, str] = {
+    "air_temp": "Air temperature",
+    "road_temp": "Road temperature",
+    "precipitation": "Precipitation type",
+    "wind_direction": "Wind direction",
+    "wind_direction_text": "Wind direction text",
+    "wind_speed": "Wind speed",
+    "wind_speed_max": "Wind speed max",
+    "humidity": "Humidity",
+    "precipitation_amount": "Precipitation amount",
+    "precipitation_amountname": "Precipitation name",
+}
+DOMAIN = "trafikverket_weatherstation"
+CONF_STATION = "station"
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_STATION): cv.string,
+        vol.Required(CONF_MONITORED_CONDITIONS, default=[]): cv.multi_select(
+            SENSOR_TYPES
+        ),
+    }
+)
+
+
+class TVWeatherConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Yale integration."""
+
+    VERSION = 1
+
+    entry: config_entries.ConfigEntry
+
+    async def async_step_import(self, config: dict):
+        """Import a configuration from config.yaml."""
+
+        self.context.update(
+            {"title_placeholders": {CONF_NAME: f"YAML import {DOMAIN}"}}
+        )
+        return await self.async_step_user(user_input=config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input[CONF_NAME]
+            api_key = user_input[CONF_API_KEY]
+            station = user_input[CONF_STATION]
+            conditions = user_input[CONF_MONITORED_CONDITIONS]
+
+            await self.async_set_unique_id(name)
+            self._abort_if_unique_id_configured()
+
+            return self.async_create_entry(
+                title=name,
+                data={
+                    CONF_NAME: name,
+                    CONF_API_KEY: api_key,
+                    CONF_STATION: station,
+                    CONF_MONITORED_CONDITIONS: conditions,
```

Comment:
An options flow to allow reconfiguring the monitored conditions would be nice in a follow-up PR

Reply:
Agree with that